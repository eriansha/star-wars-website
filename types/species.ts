export interface Species {
  id: string
  classification: string
  name: string
}
