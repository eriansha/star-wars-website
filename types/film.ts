import { Person } from '@/types/person'

export interface Film {
  id?: string
  title: string
  episodeID: number
  openingCrawl: string
  director: string
  producers: string[]
  characterConnection?: FilmCharacterConnection
}

export interface FilmCharacterConnection {
  characters: Person[]
}

export interface FilmEdge {
  node: Film
  cursor: string
}
