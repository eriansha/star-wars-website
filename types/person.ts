import { Planet } from '@/types/planet'
import { Species } from '@/types/species'

export interface Person {
  id: string
  name: string
  birthYear?: string
  eyeColor?: string
  gender?: string
  hairColor?: string
  height?: number
  masss?: number
  skinColor?: string
  homeWorld?: Planet
  species?: Species
}
