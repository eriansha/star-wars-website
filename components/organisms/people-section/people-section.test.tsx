import { cleanup, render } from '@testing-library/react'

import PeopleSection from './people-section'

import { Person } from '@/types/person'

const props: Person[] = [
  {
    id: 'abcdefg',
    name: 'Luke Sky Walker'
  },
  {
    id: 'efghijk',
    name: 'Count Dooku'
  }
]

const mockPersonCard = jest.fn()
jest.mock('@/molecules/person-card', () => () => {
  mockPersonCard(props)
  return <div>peson card</div>
})

afterEach(() => {
  cleanup
  jest.clearAllMocks()
})

describe('Person Section', () => {
  it('renders without crashing', () => {
    render(<PeopleSection people={props} />)
  })

  it('render title', () => {
    const { getByTestId } = render(<PeopleSection people={props} />)
    expect(getByTestId('title')).toHaveTextContent('People')
  })

  it('render person card as many as props', () => {
    render(<PeopleSection people={props} />)
    expect(mockPersonCard).toHaveBeenCalledTimes(2)
    expect(mockPersonCard).toHaveBeenCalledWith(props)
  })

  it('do not render person card when people props in empty or undefined', () => {
    render(<PeopleSection />)
    expect(mockPersonCard).not.toHaveBeenCalled()
  })

  it('matches snapshot', () => {
    const tree = render(<PeopleSection people={props} />)
    expect(tree.container).toMatchSnapshot()
  })
})
