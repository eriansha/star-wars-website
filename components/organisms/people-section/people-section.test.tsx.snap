// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Person Section matches snapshot 1`] = `
<div>
  <section
    class="bg-gray-dark py-5"
  >
    <h3
      class="text-2xl md:text-3xl"
      data-testid="title"
    >
      People
    </h3>
    <div
      class="flex flex-wrap gap-x-4 items-center justify-center md:justify-start"
    >
      <div>
        peson card
      </div>
      <div>
        peson card
      </div>
    </div>
  </section>
</div>
`;
